### PR TITLE
Allow shared transforms between R, Z, lambda

### LINKED ISSUE #812: Re-use transforms for Z, lambda
When `eq.sym=True`, the Z and lambda bases are the same (up to a minor difference in required derivatives), and the grids are the same as well, so the Transform for these two could point to the same object instead of 2 different objects (we would just need to set the `derivs` to the max of the derivatives needed btwn Z and lambda).

This could bring a speedup (maybe minor) to the code, and maybe even a decrease in memory usage at higher resolutions? 

And in the case `sym=False` this could be extended so that we only have one transform object used for R, Z, and lambda

### PR DESCRIPTION
**Pull request by f0uriest**
Resolves #812

### CODE REVIEW

File: desc/compute/utils.py [hunk 1]
```diff
@@ -333,13 +333,26 @@ def get_transforms(keys, obj, grid, jitable=False, **kwargs):
...
+            else:  # if we didn't exit the loop early
+                c_transform = Transform(
+                    grid,
+                    basis,
+                    derivs=derivs[c],
+                    build=False,
+                    method=method,
+                )
+            transforms[c] = c_transform
         elif c == "B":
```
Comment (rahulgaur104) [inline desc/compute/utils.py:356]: c is a key in the dictionary derivs which sounds like a dictionary of derivatives. c corresponds to the magnetic field "B". Why is the magnetic field in the dictionary containing derivatives?

Confusing naming convention. Add a few comments in the code to clarify.

**Review by rahulgaur104 — CHANGES_REQUESTED** [review]

File: desc/compute/utils.py (hunk 1 above)
Comment (dpanici) [reply desc/compute/utils.py:356]: c is meant to be just a quantity for which a transform exists, like R or Z or L. B is a transform that may exist as well (used for the boozer transform stuff). The `derivs` is a dictionary whose keys are things for which trasnforms exist, and that we may want derivs of, so I think it is natural for `"B"` to be here in its keys. `derivs` dict is explained by the docstring of `get_derivs` (https://github.com/PlasmaControl/DESC/blob/70001bcc9bc38c4e23d7d457ce5f9bb4898741ed/desc/compute/utils.py#L184)

**Review by dpanici — DISMISSED** [review]

File: desc/compute/utils.py (hunk 1 above)
Comment (f0uriest) [reply desc/compute/utils.py:356]: "B" here refers to the transform thats needed to fit the boozer harmonics

Comment (rahulgaur104) [reply desc/compute/utils.py:356]: I thought you meant the derivatives needed for the magnetic field. I'd still stay it's confusing to people who have used something like eq.compute("B"). Maybe add a comment in front of the elif in 356 saying "B means Boozer"?

**Review by rahulgaur104 — DISMISSED** [review]
Merge if all tests have passed.

**Review by dpanici — APPROVED** [review]

**Pull request merged**